platforms/apple/build-xcframework.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3
+"""
+The script builds OpenCV.framework for OSX.
+This script builds OpenCV into an xcframework compatible the platforms
+of your choice. Just run it and grab a snack; you'll be waiting a while.
+"""
+
+import sys, os, os.path, argparse, pathlib, traceback
+from cv_build_utils import execute, print_error
+
+assert sys.version_info >= (3, 6), "Python 3.6 or newer is required!"
+
+def get_or_create_folder_for_platform(platform):
+    folder_name = "./xcframework-build/%s" % platform
+    pathlib.Path(folder_name).mkdir(parents=True, exist_ok=True)
+    return folder_name
+
+def get_build_command_for_platform(platform, destination, framework_name, only_64_bit=False):
+    if platform == 'macos':
+        return "python3 ../osx/build_framework.py --archs %s --framework_name %s --without=objc --build-only-specified-archs %s" % ("x86_64,arm64", framework_name, destination)
+    elif platform == 'ios-maccatalyst':
+        return "python3 ../osx/build_framework.py --catalyst_archs %s --framework_name %s --without=objc --build-only-specified-archs %s" % ("x86_64,arm64", framework_name, destination)
+    elif platform == 'ios':
+        return "python3 ../ios/build_framework.py --iphoneos_archs %s --framework_name %s --without=objc --build-only-specified-archs %s" % ("arm64" if only_64_bit else "arm64,armv7,armv7s", framework_name, destination)
+    elif platform == 'ios-simulator':
+        return "python3 ../ios/build_framework.py --iphonesimulator_archs %s --framework_name %s --without=objc --build-only-specified-archs %s" % ("x86_64,arm64", framework_name, destination)
```

Comment:
@colejd Although I can't figure out why anyone would want it, you might as well add `i386` here:

```suggestion
        return "python3 ../ios/build_framework.py --iphonesimulator_archs %s --framework_name %s --without=objc --build-only-specified-archs %s" % ("x86_64,arm64" if only_64_bit else "i386,x86_64,arm64", framework_name, destination)
```

Reply:
👍